action.sh

```diff
@@ -1,9 +1,49 @@
 #!/usr/bin/env bash
 
+# Args:
+# $1 - inputs.working-directory
+# $2 - inputs.lint-glob
+# $3 - inputs.only-changed
+
 set -e
 
+# Get into the right directory, using the working-directory input.
 cd $GITHUB_WORKSPACE
 cd $1
 
-npm install @18f/18f-eslint@^1.1.0
-npx @18f/18f-eslint $2
\ No newline at end of file
+# Get the main branch. Historically it defaulted to "master" but it is being
+# changed to "main". We should work in both cases.
+GIT_MAIN=$(git remote show origin | awk '/HEAD branch/ {print $NF}')
+
+if [ $3 == "true" ]; then
+  echo "Only checking changed files..."
+
+  # grep returns a non-zero code if it doesn't match anything. We don't want to
+  # bail out in that case, so temporarily allow the script to continue even if
+  # there is a non-zero exit.
+  set +e
+  CHANGED=$(
+    git diff --diff-filter=ACMR --name-only --relative origin/${GIT_MAIN} -- . |
+    grep -e \.js$ -e \.jsx$ -e \.es6$
```

Comment:
These `$` without quotes worry me. Like, I know they're fine, but special characters without quotes seem to invite errors later.

Also, I _think_ this would be a little more readable as `grep -e '\.(js|jsx|esx)$'` but that's a matter of opinion

Reply:
Agreed, I thought I'd put quotes around those, but clearly I did not. 😬 

I tried that with `grep` and it just... didn't work. I don't know if that's because of Mac's funky, non-standard grep or what. 🙁

Reply:
:doh: I tested that on ubuntu with `grep -P`. nevermind about that part

Reply:
Naw, it works with `-E`, so I'm going to make the change. Looks like `-e` strips some characters of their special status and makes them literals.

Reply:
ah - `grep -E '\.(js|jsx|esx)$'` works on the old grep from OS X and the new grep that ships with ubuntu 18.04